runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java

```diff
@@ -705,6 +707,60 @@ public void testCombatBraceletUpdate()
 		assertEquals(30, slayerPlugin.getAmount());
 	}
 
+	@Test
+	public void testTaskLookupFromNPCChat() throws IOException
+	{
+		Widget npcDialog = mock(Widget.class);
+		when(npcDialog.getText()).thenReturn(TASK_EXISTING_2);
+		when(client.getWidget(WidgetInfo.DIALOG_NPC_TEXT)).thenReturn(npcDialog);
+		slayerPlugin.onGameTick(new GameTick());
+
+		net.runelite.http.api.chat.Task task = new net.runelite.http.api.chat.Task();
+		task.setTask(slayerPlugin.getTaskName());
+		task.setLocation(slayerPlugin.getTaskLocation());
+		task.setAmount(slayerPlugin.getAmount());
+		task.setInitialAmount(slayerPlugin.getInitialAmount());
+
+		when(slayerConfig.taskCommand()).thenReturn(true);
+		when(chatClient.getTask(anyString())).thenReturn(task);
+
+		ChatMessage setMessage = new ChatMessage();
+		setMessage.setType(ChatMessageType.PUBLICCHAT);
+		setMessage.setName("Adam");
+		setMessage.setMessageNode(mock(MessageNode.class));
+
+		slayerPlugin.taskLookup(setMessage, "!task");
+
+		verify(chatMessageManager).update(any(MessageNode.class));
```

Comment:
Why is this validating that the chat message manager is making changes here? This is not testing the right behavior. Instead, this should be similar to `testCheckSlayerGemKonarTask()`.

Reply:
I intended for this test to be similar to [`testTaskLookup`](https://github.com/runelite/runelite/pull/11128/files#diff-e1297a0ae6bdf66b5be1b855ea54e07bR765) in that it tests `taskLookup` does what's expected (update the chat message if we have a whitelisted task location). In this case I'm doing slightly more end-to-end testing by starting the test from the the NPC dialog, but I can remove the earlier "setup" code if we want to contain the test to just that one method.

Reply:
@Nightfirecat would this be an acceptable test given my earlier comment?

Reply:
That test is specififally verifying lookup. This one does not need to.

Reply:
I'm happy to remove these tests altogether then; I'm simply adding an entry to the locations whitelist which is used in taskLookup

Reply:
That seems like a good solution. We've got pretty sufficient tests for this code path in the plugin and just need to add the new location. :+1: